cli/tests/test_git_colocated.rs

```diff
@@ -98,28 +98,59 @@ fn test_git_colocated() {
 fn test_git_colocated_unborn_bookmark() {
     let test_env = TestEnvironment::default();
     let workspace_root = test_env.env_root().join("repo");
-    let git_repo = git2::Repository::init(&workspace_root).unwrap();
+    let git_repo = git::init(&workspace_root);
 
+    // add a file to an (in memory) index
     let add_file_to_index = |name: &str, data: &str| {
-        std::fs::write(workspace_root.join(name), data).unwrap();
-        let mut index = git_repo.index().unwrap();
-        index.add_path(Path::new(name)).unwrap();
-        index.write().unwrap();
+        let mut tree_editor = git_repo.head_tree().map(|t| t.edit().unwrap()).unwrap_or(
+            git_repo
+                .edit_tree(gix::ObjectId::empty_tree(git_repo.object_hash()))
+                .unwrap(),
+        );
+        let blob_oid = git_repo.write_blob(data).unwrap();
+        tree_editor
+            .upsert(name, gix::object::tree::EntryKind::Blob, blob_oid)
+            .unwrap();
+        let tree_id = tree_editor.write().unwrap().detach();
+        git::checkout_tree_index(&git_repo, tree_id);
     };
+
+    // checkout index (i.e., drop the in-memory changes)
     let checkout_index = || {
-        let mut index = git_repo.index().unwrap();
-        index.read(true).unwrap(); // discard in-memory cache
-        git_repo.checkout_index(Some(&mut index), None).unwrap();
+        let mut index = gix::index::File::at_or_default(
+            git_repo.index_path(),
+            git_repo.object_hash(),
+            true,
+            gix::index::decode::Options::default(),
+        )
+        .unwrap();
+        let objects = git_repo.objects.clone();
+        gix::worktree::state::checkout(
+            &mut index,
+            git_repo.work_dir().unwrap(),
+            objects,
+            &gix::progress::Discard,
+            &gix::progress::Discard,
+            &gix::interrupt::IS_INTERRUPTED,
+            gix::worktree::state::checkout::Options::default(),
+        )
+        .unwrap();
     };
 
     // Initially, HEAD isn't set.
     test_env
         .run_jj_in(&workspace_root, ["git", "init", "--git-repo", "."])
         .success();
-    assert!(git_repo.head().is_err());
+    assert!(git_repo.head().unwrap().is_unborn());
     assert_eq!(
-        git_repo.find_reference("HEAD").unwrap().symbolic_target(),
-        Some("refs/heads/master")
+        git_repo
+            .find_reference("HEAD")
+            .unwrap()
+            .target()
+            .try_name()
+            .unwrap()
```

Comment:
nit: `head_name()` also works. iirc, we had to use `.find_reference("HEAD")` in git2 because `repo.head()` would fail on unborn case.